projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { TestBed, inject } from '@angular/core/testing';
+import { COLOR_PALETTE, DataLayerColorService } from './data-layer-color.service';
+
+describe('DataLayerColorService', () => {
+  let colorService: DataLayerColorService;
+  let palette: string[] = ['#000000', '#ffffff'];
+
+  beforeEach(async () => {
+    colorService = new DataLayerColorService(palette);
+    await TestBed.configureTestingModule({
+      providers: [
+        { provide: DataLayerColorService, useValue: colorService },
+        { provide: COLOR_PALETTE, useValue: ['#000000', '#ffffff'] },
+      ],
+    }).compileComponents();
+  });
+
+  describe('chooseColorsFromPalette', () => {
+    it('should call chooseColorsFromPalette', inject([DataLayerColorService], (service: DataLayerColorService) => {
```

Comment:
```suggestion
    it('should call setColor', inject([DataLayerColorService], (service: DataLayerColorService) => {
```

Reply:
this comment still needs to be addressed. We're not expecting chooseColorsFromPalette to call itself.